### PR TITLE
RC-30121 GKE Terraform support - docs

### PR DESCRIPTION
**Pull request by poornima-rafay**

### CODE REVIEW

File: docs/resources/gke_cluster.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+- `auto_scaling` (Block List, Max: 1) Cluster autoscaler automatically creates or deletes nodes based on workload needs (see [below for nested schema](#nestedblock--spec--config--node_pools--auto_scaling))
+- `management` (Block List, Max: 1) Node management configuration (see [below for nested schema](#nestedblock--spec--config--node_pools--management))
+- `metadata` (Block List, Max: 1) Node metadata settings will be used when new nodes are created using this node pool (see [below for nested schema](#nestedblock--spec--config--node_pools--metadata))
+- `security` (Block List, Max: 1) Node security settings will be used when new nodes are created using this node pool (see [below for nested schema](#nestedblock--spec--config--node_pools--security))
+- `upgrade_settings` (Block List, Max: 1) Node pool upgrade options (see [below for nested schema](#nestedblock--spec--config--node_pools--upgrade_settings))
+
+
+<a id="nestedblock--spec--config--node_pools--machine_config"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:317]:
```suggestion
### Nested Schema for `spec.config.node_pools.machine_config`
```

File: docs/resources/gke_cluster.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+***Optional***
+
+- `boot_disk_size` (Number) Select Boot disk size. Boot disk size is permanent
+- `boot_disk_type` (String) Select Boot disk type. Storage space is less expensive for a standard persistent disk. An SSD persistent disk is better for random IOPS or for streaming throughput with low latency
+- `image_type` (String) Choose which operating system image you want to run on each node of this cluster
+- `machine_type` (String) Choose the machine type that will best fit the resource needs of your cluster
+
+
+<a id="nestedblock--spec--config--node_pools--networking"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:328]:
```suggestion
### Nested Schema for `spec.config.node_pools.networking`
```

File: docs/resources/gke_cluster.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `max_pods_per_node` (Number) This value is used to optimize the partitioning of cluster's IP address range to sub-ranges at node level
+- `network_tags` (List of String) This value is used to optimize the partitioning of cluster's IP address range to sub-ranges at node level
+
+
+<a id="nestedblock--spec--config--node_pools--node_locations"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:337]:
```suggestion
### Nested Schema for `spec.config.node_pools.node_locations`
```

File: docs/resources/gke_cluster.md [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `enabled` (Boolean) Enable providing node locations
+- `zones` (List of String) List of zones. Additional node zones must be from the same region as the original zone. Kubernetes Engine allocates the same resource footprint for each zone. The Node pool setting overrides the defaults set in Cluster basics
+
+
+<a id="nestedblock--spec--config--node_pools--security"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:346]:
```suggestion
### Nested Schema for `spec.config.node_pools.security`
```

File: docs/resources/gke_cluster.md [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `enable_integrity_monitoring` (Boolean) Integrity monitoring lets you monitor and verify the runtime boot integrity of your shielded nodes using Cloud Monitoring
+- `enable_secure_boot` (Boolean) Secure boot helps protect your nodes against boot-level and kernel-level malware and rootkits
+
+
+<a id="nestedblock--spec--config--node_pools--auto_scaling"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:355]:
```suggestion
### Nested Schema for `spec.config.node_pools.auto_scaling`
```

File: docs/resources/gke_cluster.md [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+<a id="nestedblock--spec--config--node_pools--management"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
+
+***Optional***
+
+- `auto_upgrade` (Boolean) Whether the nodes will be automatically upgraded
+
+
+<a id="nestedblock--spec--config--node_pools--metadata"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:372]:
```suggestion
### Nested Schema for `spec.config.node_pools.metadata`
```

File: docs/resources/gke_cluster.md [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `gce_instance_metadata` (Block List) Metadata to be stored in the instance (see [below for nested schema](#nestedblock--spec--config--node_pools--upgrade_settings--gce_instance_metadata))
+- `kubernetes_labels` (Block List) Use Kubernetes labels to control how workloads are scheduled to your nodes. Labels are applied to all nodes in this node pool and cannot be changed once the cluster is created (see [below for nested schema](#nestedblock--spec--config--node_pools--upgrade_settings--kubernetes_labels))
+- `node_taints` (Block List) A node taint lets you mark a node so that the scheduler avoids or prevents using it for certain Pods. Node taints can be used with tolerations to ensure that Pods aren't scheduled onto inappropriate nodes (see [below for nested schema](#nestedblock--spec--config--node_pools--upgrade_settings--node_taints))
+
+<a id="nestedblock--spec--config--node_pools--upgrade_settings--gce_instance_metadata"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings.gce_instance_metadata`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:381]:
```suggestion
### Nested Schema for `spec.config.node_pools.metadata.gce_instance_metadata`
```

File: docs/resources/gke_cluster.md [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `key` (String) Key for this metadata
+- `value` (String) Value for this metadata
+
+
+<a id="nestedblock--spec--config--node_pools--upgrade_settings--kubernetes_labels"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings.kubernetes_labels`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:390]:
```suggestion
### Nested Schema for `spec.config.node_pools.metadata.kubernetes_labels`
```

File: docs/resources/gke_cluster.md [hunk 9]
```diff
@@ -0,0 +1,458 @@
...
+
+***Optional***
+
+- `key` (String) Key for this kubernetes label
+- `value` (String) Value for this kubernetes lable
+
+
+<a id="nestedblock--spec--config--node_pools--upgrade_settings--node_taints"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings.node_taints`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:399]:
```suggestion
### Nested Schema for `spec.config.node_pools.metadata.node_taints`
```

File: docs/resources/gke_cluster.md [hunk 10]
```diff
@@ -0,0 +1,458 @@
...
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
+
+***Optional***
+
+- `max_nodes` (Number) Maximum number of nodes (per zone)
+- `min_nodes` (Number) Minimum number of nodes (per zone)
+
+
+<a id="nestedblock--spec--config--node_pools--management"></a>
+### Nested Schema for `spec.config.node_pools.upgrade_settings`
```
Comment (akshay196-rafay) [inline docs/resources/gke_cluster.md:364]:
```suggestion
### Nested Schema for `spec.config.node_pools.management`
```

**Review by akshay196-rafay — CHANGES_REQUESTED** [review]

**Comment by akshay196-rafay** [comment]
@poornima-rafay Can you share the steps your performed to generate this document? I ran `tfplugindocs generate` but result is not same as this PR.

**Comment by poornima-rafay** [comment]
> @poornima-rafay Can you share the steps your performed to generate this document? I ran `tfplugindocs generate` but result is not same as this PR.

Just run `go generate` and it produces this file but with lot of other cluster schema stuff in it. Then manually edit the fields based on what GKE resource schema has and remove the rest. Use preview to ensure the naming is correct and nesting is correct.

File: docs/resources/gke_cluster.md [hunk 11]
```diff
@@ -0,0 +1,458 @@
...
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+
+***Required***
+
+- `name` - (String) The name of the AKS Cluster. This must be unique in your organization.
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:104]: AKS should be replaced with GJE

File: docs/resources/gke_cluster.md [hunk 12]
```diff
@@ -0,0 +1,458 @@
...
+    cloud_credentials = "-dev-"
+    config {
+      gcp_project           = "dev-382813"
+      control_plane_version = "1.26"
+      location {
+        type = "zonal"
+        config {
+          zone = "us-central1-c"
+        }
+      }
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:36]: Please remove all the comments in the example. If we want to specify another example config, lets add an example in the examples directory.

File: docs/resources/gke_cluster.md [hunk 13]
```diff
@@ -0,0 +1,458 @@
...
+- `labels` (Map of String) labels of the resource
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+
+***Required***
+
+- `cloud_credentials` - (String) The name of the cloud credentials used to create and manage the cluster.
+- `config` - (Block List, Min: 1) The AKS specific cluster configuration. (See [below for nested schema](#nestedblock--spec--config))
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:120]: AKS should be replaces with GKE

File: docs/resources/gke_cluster.md [hunk 14]
```diff
@@ -0,0 +1,458 @@
...
+- `type` - (String) The GKE Cluster type. The supported value is `gke`.
+- `blueprint` (Block List, Max: 1) The blueprint to be used for this cluster. Use this for GKE cluster type (see [below for nested schema](#nestedblock--spec--blueprint))
+
+***Optional***
+
+- `proxy` (Block List, Max: 1) The proxy to be used for this cluster. Use this for GKE cluster type (see [below for nested schema](#nestedblock--spec--proxy))
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:126]: Do we need "Use this for GKE cluster type"?

File: docs/resources/gke_cluster.md [hunk 15]
```diff
@@ -0,0 +1,458 @@
...
+- `sharing` (Block List, Max: 1) Sharing spec to be used for sharing the cluster with projects (see [below for nested schema](#nestedblock--spec--sharing))
+
+<a id="nestedblock--spec--blueprint"></a>
+### Nested Schema for `spec.blueprint`
+
+***Required***
+
+- `name` (String)
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:134]: Need to add help text for name and version

File: docs/resources/gke_cluster.md [hunk 16]
```diff
@@ -0,0 +1,458 @@
...
+- `version` (String)
+
+<a id="nestedblock--spec--proxy"></a>
+### Nested Schema for `spec.proxy`
+
+Optional:
+
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:141]: We cannot have everything optional in proxy block. Need to add help text for all proxy fields

File: docs/resources/gke_cluster.md [hunk 17]
```diff
@@ -0,0 +1,458 @@
...
+- `https_proxy` (String)
+- `no_proxy` (String)
+- `proxy_auth` (String)
+
+<a id="nestedblock--spec--sharing"></a>
+### Nested Schema for `spec.sharing`
+
+Optional:
+
+- `enabled` (Boolean)
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:155]: We cannot have everything optional in sharing block. Need to add help text for all sharing fields

File: docs/resources/gke_cluster.md [hunk 18]
```diff
@@ -0,0 +1,458 @@
...
+- `projects` (Block List) (see [below for nested schema](#nestedblock--spec--sharing--projects))
+
+<a id="nestedblock--spec--sharing--projects"></a>
+### Nested Schema for `spec.sharing.projects`
+
+Optional:
+
+- `name` (String)
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:163]: Need to add help text

File: docs/resources/gke_cluster.md [hunk 19]
```diff
@@ -0,0 +1,458 @@
...
+- `control_plane_version` (String) Kubernetes version of ControlPlane
+- `location` (Block List, Max: 1) GKE cluster location configuration. (see [below for nested schema](#nestedblock--spec--config--location))
+- `network` (Block List, Max: 1) GKE cluster Network configuration. (see [below for nested schema](#nestedblock--spec--config--network))
+- `node_pools` (Block List) GKE cluster node pool configuration. (see [below for nested schema](#nestedblock--spec--config--node_pools))
+
+Optional:
+
+- `features` (Block List, Max: 1) GKE cluster additional features configuration. (see [below for nested schema](#nestedblock--spec--config--features))
+- `pre_bootstrap_commands` (List of String) Commands will be executed every time Cluster nodes come up. Example: Node Creation, Node Restart.
+- `security` (Block List, Max: 1) GKE cluster Network configuration. (see [below for nested schema](#nestedblock--spec--config--security))
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:181]: Help text needs update

File: docs/resources/gke_cluster.md [hunk 20]
```diff
@@ -0,0 +1,458 @@
...
+
+- `type` (String) GKE Cluster location can be either `zonal` or `regional`
+- `config` (Block List, Max: 1) (see [below for nested schema](#nestedblock--spec--config--location--config))
+
+***Optional***
+
+- `default_node_locations` (Block List, Max: 1) For Zonal cluster increase availability by selecting more than one zone. For Regional cluster By default, Kubernetes Engine runs nodes of a regional cluster across three zones within a region. Select this option if you want to manually specify the zones in which this cluster's nodes run. All zones must be within the same region. (see [below for nested schema](#nestedblock--spec--config--location--default_node_locations))
+
+<a id="nestedblock--spec--config--location--config"></a>
+### Nested Schema for `spec.config.location.type`
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:197]: spec.config.location.type should be spec.config.location.config

File: docs/resources/gke_cluster.md [hunk 21]
```diff
@@ -0,0 +1,458 @@
...
+
+- `region` (String) Regional location in which the cluster's control plane and nodes are located. For ZonalCluster only zone information is sufficient. For Regional Cluster, both region and zone should be provided.
+- `zone` (String) Zone in the region where bootstrap VM is created for cluster provisioning. For ZonalCluster only zone information is sufficient. For Regional
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:200]: zone is used only for bootstrap vm?

**Review by kutumba-rafay — CHANGES_REQUESTED** [review]

File: docs/resources/gke_cluster.md (hunk 1 above)
Comment (poornima-rafay) [reply docs/resources/gke_cluster.md]: changed.

File: docs/resources/gke_cluster.md (hunk 3 above)
Comment (poornima-rafay) [reply docs/resources/gke_cluster.md]: changed

File: docs/resources/gke_cluster.md (hunk 21 above)
Comment (poornima-rafay) [reply docs/resources/gke_cluster.md]: @kutumba-rafay Yes. When cluster type is `Regional`, we force user to provide a zone in that region and this zone is where we bring up bootstrap VM. Zone info is not used for anything else for regional cluster. 
Updated text. Hopefully it is clear now.

**Review by ankur-rafay — APPROVED** [review]

File: docs/resources/gke_cluster.md [hunk 22]
```diff
@@ -0,0 +1,564 @@
...
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+
+***Required***
+
+- `cloud_credentials` - (String) The name of the cloud credentials used to create and manage the cluster.
+- `config` - (Block List, Min: 1) The GKE specific cluster configuration. (See [below for nested schema](#nestedblock--spec--config))
+- `type` - (String) The GKE Cluster type. The supported value is `gke`.
```
Comment (kutumba-rafay) [inline docs/resources/gke_cluster.md:226]: change it to "Cluster type"

**Review by kutumba-rafay — CHANGES_REQUESTED** [review]

**Review by akshay196-rafay — APPROVED** [review]

**Review by kutumba-rafay — APPROVED** [review]

**Pull request merged**